201-site-to-site-vpn-fqdn-bgp/azuredeploy.json

```diff
@@ -0,0 +1,1109 @@
+    "subnet21Ref": "[resourceId('Microsoft.Network/virtualNetworks/subnets',variables('vNet2Name'), variables('subnet21Name') )]",
+    "gateway1Name": "[parameters('gateway1Name')]",
+    "gateway2Name": "[parameters('gateway2Name')]",
+    "gateway1PublicIP1Name": "[parameters('gateway1PublicIP1Name')]",
+    "gateway1PublicIP2Name": "[parameters('gateway1PublicIP2Name')]",
+    "gateway2PublicIP1Name": "[parameters('gateway2PublicIP1Name')]",
+    "gateway2PublicIP2Name": "[parameters('gateway2PublicIP2Name')]",
+    "gateway2PublicIP1Id": "[resourceId('Microsoft.Network/publicIPAddresses',variables('gateway2PublicIP1Name'))]",
+    "gateway2PublicIP2Id": "[resourceId('Microsoft.Network/publicIPAddresses',variables('gateway2PublicIP2Name'))]",
+    "gateway1PublicIP1Id": "[resourceId('Microsoft.Network/publicIPAddresses',variables('gateway1PublicIP1Name'))]",
+    "gateway1PublicIP2Id": "[resourceId('Microsoft.Network/publicIPAddresses',variables('gateway1PublicIP2Name'))]",
+    "dnsLabelgtw1PubIP1": "[parameters('dnsLabelgtw1PubIP1')]",
+    "dnsLabelgtw1PubIP2": "[parameters('dnsLabelgtw1PubIP2')]",
+    "dnsLabelgtw2PubIP1": "[parameters('dnsLabelgtw2PubIP1')]",
+    "dnsLabelgtw2PubIP2": "[parameters('dnsLabelgtw2PubIP2')]",
+    "asnGtw1": "[parameters('asnGtw1')]",
+    "asnGtw2": "[parameters('asnGtw2')]",
+    "localGatewayName11": "[parameters('localGatewayName11')]",
+    "localGatewayName12": "[parameters('localGatewayName12')]",
+    "localGatewayName21": "[parameters('localGatewayName21')]",
+    "localGatewayName22": "[parameters('localGatewayName22')]",
+    "connectionName11-21": "[parameters('connectionName11-21')]",
+    "connectionName12-22": "[parameters('connectionName12-22')]",
+    "connectionName21-11": "[parameters('connectionName21-11')]",
+    "connectionName22-12": "[parameters('connectionName22-12')]",
+    "vpnGatewayGeneration": "[parameters('vpnGatewayGeneration')]",
+    "vm1Name": "[parameters('vm1Name')]",
+    "vm2Name": "[parameters('vm2Name')]",
+    "nsg1": "[concat(parameters('prefix'),'nsg1')]",
+    "nsg2": "[concat(parameters('prefix'),'nsg2')]",
+    "nic1Name": "[concat(variables('vm1Name'), '-nic')]",
+    "nic2Name": "[concat(variables('vm2Name'), '-nic')]",
+    "imagePublisher": "[parameters('imagePublisher')]",
+    "imageOffer": "[parameters('imageOffer')]",
+    "imageSKU": "[parameters('imageSKU')]",
+    "vm1PublicIPName": "[concat(variables('vm1Name'), '-pubIP')]",
+    "vm2PublicIPName": "[concat(variables('vm2Name'), '-pubIP')]",
+    "vmSize": "[parameters('vmSize')]",
+    "storageAccountType": "[parameters('storageAccountType')]"
+  },
+  "resources": [
+    {
+
+      "type": "Microsoft.Network/networkSecurityGroups",
+      "name": "[variables('nsg1')]",
+      "apiVersion": "2020-08-01",
+      "location": "[variables('location1')]",
+      "comments": "network security group applied to the subnets in the VNet1",
```

Comment:
See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#sort-order-of-properties for the proper sort order of properties

Reply:
I have changed the proper sort order as in best practice